pkg/kube/client.go

```diff
@@ -105,6 +109,64 @@ func (c *Client) Create(resources ResourceList) (*Result, error) {
 	return &Result{Created: resources}, nil
 }
 
+func transformRequests(req *rest.Request) {
+	group := metav1beta1.GroupName
+	version := metav1beta1.SchemeGroupVersion.Version
+
+	tableParam := fmt.Sprintf("application/json;as=Table;v=%s;g=%s, application/json", version, group)
+	req.SetHeader("Accept", tableParam)
+
+	// if sorting, ensure we receive the full object in order to introspect its fields via jsonpath
+	req.Param("includeObject", "Object")
+}
+
+func (c *Client) Get(resources ResourceList) (string, error) {
```

Comment:
If this is used with the normal display this would be fine. But, if JSON or YAML is the output the complex string is going to end up inside a key and not be easily parsed.

I would suggest getting the output should be different per type and a print time operation depending on the output format. This would be more useful for getting this information in a machine format.

Reply:
Hi Matt,

Commited code is worked fine for different output formats like yaml and json as well.

Below is the genearted output without format and with different format cases

**Without output format case**

helm status firsthelm-0-1579514513 
```
NAME: firsthelm-0-1579514513
LAST DEPLOYED: Mon Jan 20 05:01:57 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
RESOURCES:
==> v1/Service
NAME                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
firsthelm-0-1579514513   ClusterIP   10.96.41.237   <none>        80/TCP    50d

==> v1/Deployment
NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
firsthelm-0-1579514513   1/1     1            1           50d


NOTES:
1. Get the application URL by running these commands:
  export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=firsthelm,app.kubernetes.io/instance=firsthelm-0-1579514513" -o jsonpath="{.items[0].metadata.name}")
  echo "Visit http://127.0.0.1:8080 to use your application"
  kubectl port-forward $POD_NAME 8080:80
```

**With yaml as output format**

helm status firsthelm-0-1579514513 **-o yaml**

```
Resources: |2+

  ==> v1/Service
  NAME                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
  firsthelm-0-1579514513   ClusterIP   10.96.41.237   <none>        80/TCP    50d

  ==> v1/Deployment
  NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
  firsthelm-0-1579514513   1/1     1            1           50d

hooks:
- events:
  - test
  kind: Pod
  last_run:
    completed_at: ""
    phase: ""
    started_at: ""
  manifest: |-
    apiVersion: v1
    kind: Pod
    metadata:
      name: "firsthelm-0-1579514513-test-connection"
      labels:
        app.kubernetes.io/name: firsthelm
        helm.sh/chart: firsthelm-0.1.0
        app.kubernetes.io/instance: firsthelm-0-1579514513
        app.kubernetes.io/version: "1.0"
        app.kubernetes.io/managed-by: Helm
      annotations:
        "helm.sh/hook": test-success
    spec:
      containers:
        - name: wget
          image: busybox
          command: ['wget']
          args:  ['firsthelm-0-1579514513:80']
      restartPolicy: Never
  name: firsthelm-0-1579514513-test-connection
  path: firsthelm/templates/tests/test-connection.yaml
info:
  deleted: ""
  description: Install complete
  first_deployed: "2020-01-20T05:01:57.370835613-05:00"
  last_deployed: "2020-01-20T05:01:57.370835613-05:00"
  notes: |
    1. Get the application URL by running these commands:
      export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=firsthelm,app.kubernetes.io/instance=firsthelm-0-1579514513" -o jsonpath="{.items[0].metadata.name}")
      echo "Visit http://127.0.0.1:8080 to use your application"
      kubectl port-forward $POD_NAME 8080:80
  status: deployed
manifest: |
  ---
  # Source: firsthelm/templates/service.yaml
  apiVersion: v1
  kind: Service
  metadata:
    name: firsthelm-0-1579514513
    labels:
      app.kubernetes.io/name: firsthelm
      helm.sh/chart: firsthelm-0.1.0
      app.kubernetes.io/instance: firsthelm-0-1579514513
      app.kubernetes.io/version: "1.0"
      app.kubernetes.io/managed-by: Helm
  spec:
    type: ClusterIP
    ports:
      - port: 80
        targetPort: http
        protocol: TCP
        name: http
    selector:
      app.kubernetes.io/name: firsthelm
      app.kubernetes.io/instance: firsthelm-0-1579514513
  ---
  # Source: firsthelm/templates/deployment.yaml
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: firsthelm-0-1579514513
    labels:
      app.kubernetes.io/name: firsthelm
      helm.sh/chart: firsthelm-0.1.0
      app.kubernetes.io/instance: firsthelm-0-1579514513
      app.kubernetes.io/version: "1.0"
      app.kubernetes.io/managed-by: Helm
  spec:
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/name: firsthelm
        app.kubernetes.io/instance: firsthelm-0-1579514513
    template:
      metadata:
        labels:
          app.kubernetes.io/name: firsthelm
          app.kubernetes.io/instance: firsthelm-0-1579514513
      spec:
        containers:
          - name: firsthelm
            image: "nginx:stable"
            imagePullPolicy: IfNotPresent
            ports:
              - name: http
                containerPort: 80
                protocol: TCP
            livenessProbe:
              httpGet:
                path: /
                port: http
            readinessProbe:
              httpGet:
                path: /
                port: http
            resources:
              {}
name: firsthelm-0-1579514513
namespace: default
version: 1
```

**With json as output format**

helm status firsthelm-0-1579514513  **-o json**
```
{"name":"firsthelm-0-1579514513","info":{"first_deployed":"2020-01-20T05:01:57.370835613-05:00","last_deployed":"2020-01-20T05:01:57.370835613-05:00","deleted":"","description":"Install complete","status":"deployed","notes":"1. Get the application URL by running these commands:\n  export POD_NAME=$(kubectl get pods --namespace default -l \"app.kubernetes.io/name=firsthelm,app.kubernetes.io/instance=firsthelm-0-1579514513\" -o jsonpath=\"{.items[0].metadata.name}\")\n  echo \"Visit http://127.0.0.1:8080 to use your application\"\n  kubectl port-forward $POD_NAME 8080:80\n"},"manifest":"---\n# Source: firsthelm/templates/service.yaml\napiVersion: v1\nkind: Service\nmetadata:\n  name: firsthelm-0-1579514513\n  labels:\n    app.kubernetes.io/name: firsthelm\n    helm.sh/chart: firsthelm-0.1.0\n    app.kubernetes.io/instance: firsthelm-0-1579514513\n    app.kubernetes.io/version: \"1.0\"\n    app.kubernetes.io/managed-by: Helm\nspec:\n  type: ClusterIP\n  ports:\n    - port: 80\n      targetPort: http\n      protocol: TCP\n      name: http\n  selector:\n    app.kubernetes.io/name: firsthelm\n    app.kubernetes.io/instance: firsthelm-0-1579514513\n---\n# Source: firsthelm/templates/deployment.yaml\napiVersion: apps/v1\nkind: Deployment\nmetadata:\n  name: firsthelm-0-1579514513\n  labels:\n    app.kubernetes.io/name: firsthelm\n    helm.sh/chart: firsthelm-0.1.0\n    app.kubernetes.io/instance: firsthelm-0-1579514513\n    app.kubernetes.io/version: \"1.0\"\n    app.kubernetes.io/managed-by: Helm\nspec:\n  replicas: 1\n  selector:\n    matchLabels:\n      app.kubernetes.io/name: firsthelm\n      app.kubernetes.io/instance: firsthelm-0-1579514513\n  template:\n    metadata:\n      labels:\n        app.kubernetes.io/name: firsthelm\n        app.kubernetes.io/instance: firsthelm-0-1579514513\n    spec:\n      containers:\n        - name: firsthelm\n          image: \"nginx:stable\"\n          imagePullPolicy: IfNotPresent\n          ports:\n            - name: http\n              containerPort: 80\n              protocol: TCP\n          livenessProbe:\n            httpGet:\n              path: /\n              port: http\n          readinessProbe:\n            httpGet:\n              path: /\n              port: http\n          resources:\n            {}\n","hooks":[{"name":"firsthelm-0-1579514513-test-connection","kind":"Pod","path":"firsthelm/templates/tests/test-connection.yaml","manifest":"apiVersion: v1\nkind: Pod\nmetadata:\n  name: \"firsthelm-0-1579514513-test-connection\"\n  labels:\n    app.kubernetes.io/name: firsthelm\n    helm.sh/chart: firsthelm-0.1.0\n    app.kubernetes.io/instance: firsthelm-0-1579514513\n    app.kubernetes.io/version: \"1.0\"\n    app.kubernetes.io/managed-by: Helm\n  annotations:\n    \"helm.sh/hook\": test-success\nspec:\n  containers:\n    - name: wget\n      image: busybox\n      command: ['wget']\n      args:  ['firsthelm-0-1579514513:80']\n  restartPolicy: Never","events":["test"],"last_run":{"started_at":"","completed_at":"","phase":""}}],"version":1,"namespace":"default","Resources":"\n==\u003e v1/Service\nNAME                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE\nfirsthelm-0-1579514513   ClusterIP   10.96.41.237   \u003cnone\u003e        80/TCP    50d\n\n==\u003e v1/Deployment\nNAME                     READY   UP-TO-DATE   AVAILABLE   AGE\nfirsthelm-0-1579514513   1/1     1            1           50d\n\n"}
```


I think that, in helm2 also: "helm status" command with different output formats, the output is same.

Reply:
The release part of this is a string of the output. Instead, we should have that as structured metadata. It doesn't need to be the same as the output of `kubectl get po --output json`. I would expect the structure to be a JSON object that represents the table as JSON.

Right now (in Helm v2) machines need to use a regex to parse the content of the status information. That includes getting information about things like pods. If this was represented as JSON/YAML when the output was JSON or YAML it would make machines using this easier.

Reply:
@sarvaniy this seems to require fixing, are you able to do that?